src/icloudpd/base.py

```diff
@@ -418,6 +452,18 @@ def download_photo_(counter: Counter, photo: PhotoAsset) -> bool:
                     photo.created)
                 created_date = photo.created
 
+            if created_before and created_date > created_before:

```

Comment:
These are the conditions we need to validate with a test. Parameter absence is tested, but comparison is not completely validated IMO - need to check we indeed leave some assets and filter out others (assets created on the same date as parameter is the case that should be present in tests as well). In e2e tests that will be translated into checking that some files are downloaded and some are not.